packages/eslint-plugin-material-ui/src/rules/restricted-path-imports.test.js

```diff
@@ -12,6 +12,7 @@ ruleTester.run('restricted-path-imports', rule, {
     "import { blue } from '@material-ui/core/colors'",
     "import * as colors from '@material-ui/core/colors'",
     "import * as colors from '@another/core/styles/withStyles'",
+    "import describeConformance from '@material-ui/core/test-utils/describeConformance'",
```

Comment:
Do we still use this path or why should this be valid?

Reply:
Ah not sure, I just copied this lint rule into my repo at work cos it's not published, and noticed this inconsistency.

Reply:
But I think you do:

https://github.com/mui-org/material-ui/blob/4fba0dafd30f608937efa32883d151ba01fc9681/packages/material-ui-lab/src/Alert/Alert.test.js#L3

Reply:
Tricky. This was never intended to be a public module which is why it is not exported from `test-utils/index`. But let's leave it be for now (since it didn't cause us any trouble after we hid it in a deep import).

We should add a `TODO` notice that we this should fail in v5 where we'll move it to our internal `/test/utils`.

Reply:
Sorry, that's not what I meant - I copied and pasted the lint rule along with its tests (just in case ESLint 7 comes out and breaks it) into my repo to guard against deep imports in our codebase (I'm trying to make our UMD build depend on yours which means no deep imports and actually led me to https://github.com/mui-org/material-ui/issues/20353), and then as I was familiarising myself with your code & tests, I saw that the condition wasn't being tested :) We don't use this util in our tests, but I'm happy to add the `TODO`.

Reply:
Do you just want

```js
// TODO v5: move to internal `/test/utils`.
```

at the top of `packages/material-ui/src/test-utils/describeConformance.js`? Given it's not exported from `index.js` and is a 3rd level import, I'd say it's only fair because it's not deemed public as per https://material-ui.com/guides/minimizing-bundle-size/#option-1.

Btw, any plans to make this lint rule public too?

Reply:
Oh sorry, I was taking mental notes :smile: I was thinking about just moving it now but that would just potentially break code unnencessarily. But a `TODO` we need to add while we're at it.

Reply:
> We should add a TODO notice that we this should fail in v5 where we'll move it to our internal /test/utils.

@eps1lon I think that it could be interesting to have this module public, especially for third-party authors that went to enforce the same constraints than our components. For instance, if we didn't plan to move `@material-ui/pickers` into the mono-repository, it would have helped.

Reply:
Sure but this should be another package so that we can properly declare (peer) dependencies. Right now you have to be able to understand the source to be able to make it work. At that point you might as well inline it.

Reply:
@eps1lon Oh, so like a `@material-ui/test` package? Maybe it's simpler to make it private for now?

Reply:
As I said there's no gain now by making this private now. Let's keep this simple and mark this as undesired and strip out test-utils in v5. We can discuss what's useful in test-utils for other packages in a separate issue (and if this should be a "test" package).

Reply:
How would you like me to proceed with this?

1. add a `TODO` to the top of the `describeConformance` module
2. point 1 but also re-export it from `core/test-utils`, update imports, and remove the exclusion from the ESLint rule